pkg/collector/corechecks/system/disk/disk/disk_nix.go

```diff
@@ -4,149 +4,168 @@
 // Copyright 2016-present Datadog, Inc.
 
 //go:build !windows
+// +build !windows
 
+//nolint:revive // TODO(AGENTRUN) Fix revive linter
 package disk
 
 import (
-	"fmt"
-	"path/filepath"
+	"bytes"
+	"encoding/xml"
+	"io"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
 
-	"github.com/shirou/gopsutil/v4/disk"
-
-	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
-	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
-	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-// for testing
-var (
-	diskPartitions = disk.Partitions
-	diskUsage      = disk.Usage
-
-	ioCounters = disk.IOCounters
-)
-
-// Check stores disk-specific additional fields
-type Check struct {
-	core.CheckBase
-	cfg *diskConfig
+var LsblkCommand = func() (string, error) {
+	cmd := exec.Command("lsblk", "--noheadings", "--raw", "--output=NAME,LABEL")
+	output, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	return string(output), nil
 }
 
-// Run executes the check
-func (c *Check) Run() error {
-	sender, err := c.GetSender()
+func (c *Check) fetchAllDeviceLabelsFromLsblk() error {
+	log.Debugf("Fetching all device labels from lsblk")
+	rawOutput, err := LsblkCommand()
 	if err != nil {
 		return err
 	}
+	log.Debugf("lsblk output: %s", rawOutput)
+	lines := strings.Split(strings.TrimSpace(string(rawOutput)), "\n")
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		log.Debugf("processing line: '%s'", line)
+		if line == "" {
+			continue
+		}
+		// Typically line looks like:
+		// sda1  MY_LABEL
+		fields := strings.Fields(line)
+		if len(fields) < 2 {
+			// skip malformed lines
+			continue
+		}
+		device := "/dev/" + fields[0]
+		label := fields[1]
+		c.deviceLabels[device] = label
```

Comment:
Should we be worried about overwriting values on the map?

Reply:
We seems to create a new map for `fetchAllDeviceLabelsFromBlkidCache` and `fetchAllDeviceLabelsFromBlkid ` any reason why we not the same thing for this function?

Reply:
Fixed